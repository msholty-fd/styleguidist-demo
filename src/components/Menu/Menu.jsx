import React from 'react'
import PropTypes from 'prop-types'
import MaterialMenu from 'material-ui/Menu'
import getMuiTheme from 'material-ui/styles/getMuiTheme'

import list from '../../styles/components/list'
import menu from '../../styles/components/menu'
import theme from '../../styles/Theme'

const getStyles = props => ({
  root: Object.assign({}, menu, props.style),
  list: Object.assign({}, list, props.listStyle)
})

export default class Menu extends React.Component {
  static propTypes = MaterialMenu.propTypes;

  static childContextTypes = {
    muiTheme: PropTypes.object.isRequired
  };

  getChildContext () {
    return {muiTheme: getMuiTheme(theme)}
  };

  render () {
    const styles = getStyles(this.props)
    const { children, ...others } = this.props

    return (
      <MaterialMenu
        {...others}
        style={styles.root}
        listStyle={styles.list}
      >
        {children}
      </MaterialMenu>
    )
  }
}
