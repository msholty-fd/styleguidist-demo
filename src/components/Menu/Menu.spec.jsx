import React from 'react'
import renderer from 'react-test-renderer'
import { MenuItem } from 'material-ui'
import { describe, it, expect } from 'global'

import Menu from './Menu'

jest.mock('material-ui/Menu', () => 'Menu')
jest.mock('material-ui/MenuItem', () => 'MenuItem')

const component = (
  <Menu>
    <MenuItem value={1} primaryText='Never' />
    <MenuItem value={2} primaryText='Every Night' />
    <MenuItem value={3} primaryText='Weeknights' />
    <MenuItem value={4} primaryText='Weekends' />
    <MenuItem value={5} primaryText='Weekly' />
  </Menu>
)

describe('<Menu />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(component)
    const tree = renderedComponent.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
