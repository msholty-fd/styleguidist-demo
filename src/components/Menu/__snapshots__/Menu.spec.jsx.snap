// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match the snapshot 1`] = `
<Menu
  listStyle={
    Object {
      "borderRadius": 5,
      "padding": "10px 0",
    }
  }
  style={
    Object {
      "marginTop": 5,
    }
  }
>
  <MenuItem
    primaryText="Never"
    value={1}
  />
  <MenuItem
    primaryText="Every Night"
    value={2}
  />
  <MenuItem
    primaryText="Weeknights"
    value={3}
  />
  <MenuItem
    primaryText="Weekends"
    value={4}
  />
  <MenuItem
    primaryText="Weekly"
    value={5}
  />
</Menu>
`;
